common/components/componentsBySection/FindProjects/FavoriteFilter.jsx

```diff
@@ -0,0 +1,59 @@
+// @flow
+
+import React from "react";
+import { Container } from "flux/utils";
+import IconToggle from "../../chrome/IconToggle.jsx";
+import { GlyphStyles, GlyphSizes } from "../../utils/glyphs.js";
+import ProjectSearchStore from "../../stores/ProjectSearchStore.js";
+import ProjectSearchDispatcher from "../../stores/ProjectSearchDispatcher.js";
+import metrics from "../../utils/metrics.js";
+
+type State = {|
+  favoritesOnly: boolean,
+|};
+
+class FavoriteFilter extends React.Component<{||}, State> {
+  constructor(props: Props): void {
+    super(props);
+    this.state = {
+      favoritesOnly: false,
+    };
+  }
+
+  static getStores(): $ReadOnlyArray<FluxReduceStore> {
+    return [ProjectSearchStore];
+  }
+
+  static calculateState(prevState: State): State {
+    const state: State = {
+      favoritesOnly: ProjectSearchStore.getFavoritesOnly(),
+    };
+
+    return state;
+  }
+
+  doToggle(): void {
+    const newFavoritesOnly: boolean = !this.state.favoritesOnly;
+    ProjectSearchDispatcher.dispatch({
+      type: "SET_FAVORITES_ONLY",
+      favoritesOnly: newFavoritesOnly,
+    });
+    metrics.logFilterProjectsByFavorite(newFavoritesOnly);
+  }
+
+  render(): React$Node {
+    return (
+      <div className="favorite-filter" onClick={this.doToggle.bind(this)}>
+        Favorites
+        <IconToggle
+          toggled={this.state.favoritesOnly}
+          toggleOnIcon={GlyphStyles.HeartFilled}
+          toggleOffIcon={GlyphStyles.HeartEmpty}
+          size={GlyphSizes.X2}
+        />
+      </div>
```

Comment:
Technically, the spec for the filter in the nav has the favorite heart always being filled in, but we then lose the visual cue that favorites are enabled as a filter. We should ask design how they want it to be made known if favorite filtering is enabled. Maybe for now something like this mockup?
![image](https://user-images.githubusercontent.com/16870466/134578310-e9d0e397-e636-42ba-b301-ef8277302eb6.png)

My thought is if we're putting favorites in the filter area because it's a filter, we need to make it a filter all the way and that means having the pill show up when the favorites filter is active. What do you think?